cmake/modules/FindBabbleSim.cmake

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# FindBabbleSim module for locating BabbleSim
+#
+# The module defines the following variables:
+#
+# 'BSIM_COMPONENTS_PATH'
+# Path to the BabbleSim components source folder
+#
+# 'BSIM_OUT_PATH'
+# Path to the BabbleSim build output root path (under which libraries and binaries) are kept
+#
+# We first try to find it via the environment variables BSIM_OUT_PATH and BSIM_COMPONENTS_PATH.
+# If these are not set, as a fallback we attempt to find it thru west, in case the user
+# fetched babblesim using the manifest.
+# Note that what we find thru the environment variables is meant to have precedence.
+#
+# If BabbleSim cannot be found we error right away with a message trying to guide users
+
+zephyr_get(BSIM_COMPONENTS_PATH)
+zephyr_get(BSIM_OUT_PATH)
+
+if ((DEFINED WEST) AND (NOT DEFINED BSIM_COMPONENTS_PATH) AND (NOT DEFINED BSIM_OUT_PATH))
+  # Let's ask west for the bsim_project existence and its path
+  execute_process(COMMAND ${WEST}
+    status babblesim_base
+    OUTPUT_QUIET
+    RESULT_VARIABLE ret_val1)
+  execute_process(COMMAND ${WEST}
+    list babblesim_base -f {posixpath}
+    OUTPUT_VARIABLE BSIM_BASE_PATH
+    RESULT_VARIABLE ret_val2)
+  if (NOT (${ret_val1} OR ${ret_val2}))
+    string(STRIP ${BSIM_BASE_PATH} BSIM_COMPONENTS_PATH)
+    get_filename_component(BSIM_OUT_PATH ${BSIM_COMPONENTS_PATH}/.. ABSOLUTE)
+  endif()
+endif()
+
+message(STATUS "Using BSIM from BSIM_COMPONENTS_PATH=${BSIM_COMPONENTS_PATH}\
+ BSIM_OUT_PATH=${BSIM_OUT_PATH}")
```

Comment:
Probably best to print this out after the bit below so it only outputs if it's found